app/src/main/res/layout/activity_main.xml

```diff
@@ -1,18 +1,16 @@
 <?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
+
+<LinearLayout
```

Comment:
we can come back to this, but this is a great example of where we can do one of the following: 
1. Use the recycler view as the topmost view
2. Use a framelayout instead of linear. 

Ultimately, it doesn't matter for you. This linear layout works just fine. The above judgements would be made to choose a layout that is most performant. Ultimately, it doesn't matter though and you will pick that up over time and knowing which ones to use where :)

Reply:
sounds good. I had just the recycler view there initially like we talked about yesterday, but it wasn't liking it for some reason and the tutorial I was looking mentioned a linear layout. maybe I didn't have the declaration in yet or something, but yea can chat more on it to understand best practice.

Reply:
LinearLayout totally works, and tbh this is a nitpick for something like this. Could be a host of reasons why it was yelling at you with it at as the upper most. 

Once you get more familiar with the ecosystem you will know what layouts to use where based on the use case - similar to how you do it today for iOS :)